deepchecks/nlp/text_data.py

```diff
@@ -0,0 +1,583 @@
+    label : t.Optional[TTextLabel], default: None
+        The label for the text data. Can be either a text_classification label or a token_classification label.
+        If None, the label is not set.
+    task_type : str, default: None
+        The task type for the text data. Can be either 'text_classification' or 'token_classification'. Must be set if
+        label is provided.
+    label_name : t.Optional[str], default: None
+        The name of the label column. If None, the label column name is 'label'.
+    dataset_name : t.Optional[str], default: None
+        The name of the dataset. If None, the dataset name will be defined when running it within a check.
+    index: t.Optional[t.Sequence[int]], default: None
+        The index of the samples. If None, the index is set to np.arange(len(raw_text)).
+    """
+
+    _dataset: TTextDataset
+    _task_type: t.Optional[TaskType]
+    _has_label: bool
+    _label_name: str
+    _is_multilabel: bool = False
+    _classes: t.Optional[t.List[t.Union[str, int]]] = None
+    name: t.Optional[str] = None
+
+    def __init__(
+            self,
+            raw_text: t.Sequence[str],
+            label: t.Optional[TTextLabel] = None,
+            task_type: str = None,
+            label_name: t.Optional[str] = None,
+            dataset_name: t.Optional[str] = None,
+            index: t.Optional[t.Sequence[int]] = None,
+    ):
+
+        # Require explicitly setting task type if label is provided
+        if task_type is None:
+            if self.label is not None:
+                raise DeepchecksValueError('task_type must be set when label is provided')
+            self._task_type = None
+        elif task_type == 'text_classification':
+            self._task_type = TaskType.TEXT_CLASSIFICATION
+        elif task_type == 'token_classification':
+            self._task_type = TaskType.TOKEN_CLASSIFICATION
+        else:
+            raise DeepchecksNotSupportedError(f'task_type {task_type} is not supported, must be one of '
+                                              f'text_classification, token_classification')
+
+        # Set label name to default if not provided
+        if label_name is None:
+            self._label_name = DEFAULT_LABEL_NAME
+        elif isinstance(label_name, (str, float, int)):
+            self._label_name = label_name
+        else:
+            raise DeepchecksNotSupportedError(f'label_name {label_name} is not supported, must be one of '
+                                              f'str, float, int')
+
+        self._validate_text(raw_text)
+        label = self._validate_and_process_label(label, raw_text)
+
+        if index is None:
+            index = np.arange(len(raw_text))
+        elif len(index) != len(raw_text):
+            raise DeepchecksValueError('index must be the same length as raw_text')
+
+        # Create dataset
+        self._dataset = {'text': raw_text, self._label_name: label, 'index': index}
```

Comment:
IMO more clear + easy for further use to save each of them in a different variable i.e
self.label , self.raw_data
Also save them as array so the index argument is redundant